iopipe/contrib/logger/request.py

```diff
@@ -11,21 +11,26 @@
 logger = logging.getLogger(__name__)
 
 
-def upload_log_data(url, stream_or_file):
+def upload_log_data(url, stream_or_file, config):
     """
     Uploads log data to IOpipe.
 
     :param url: The signed URL
     :param stream_or_file: The log data stream or file
+    :param config: The IOpipe config
     """
     try:
         logger.debug("Uploading log data to IOpipe")
         if isinstance(stream_or_file, StringIO):
             stream_or_file.seek(0)
-            response = requests.put(url, data=stream_or_file, timeout=None)
+            response = requests.put(
+                url, data=stream_or_file, timeout=config["network_timeout"]
```

Comment:
Hmm, do we want to use the same network timeout as collector? Or have it be a separate config item? The default value for this is 5000ms and I think we can be much more aggressive with the signer API (100ms)

Reply:
(honestly, could be that we should make that 5000 much smaller)

Reply:
I don't think the service characteristics of the collector and signer are different enough to warrant a significantly different timeout. If both services are healthy these timeouts should never be hit. The timeout acts as a n upper bound, or worst case, so would be conservative here. I'm not even sure 5s is going to be enough for S3 uploads.